server/src/main/java/org/elasticsearch/ingest/Processor.java

```diff
@@ -122,6 +123,8 @@ Processor create(Map<String, Processor.Factory> processorFactories, String tag,
 
         public final IngestService ingestService;
 
+        public final ThreadPool threadPool;
```

Comment:
In the past we tried to not directly expose `ThreadPool` to avoid plugins to just have access to this internal service. This was part of a bigger idea that plugins shouldn't have direct access to internal services. This is why the `Parameters` class has the `threadContext`, `relativeTimeSupplier` and `scheduler` fields separately instead of just exposing a `threadPool` field in this class. 

Maybe we can try to keep this? We can expose a `Consumer<Runnable>` field invokes `threadPool.generic().execute(...)` with the provided Runnable instance. 

Or we can reuse the `Parameters#scheduler` field? And then provide a delay of `0` ms. This should also execute  the provided runnable on a the generic threadpool.

Reply:
Agreed to not expose `ThreadPool` (thanks for catching that).  I exposed a Consumer<Runnable>, can you take another look.

Reply:
👍 LGTM.